lpc55_areas/src/lib.rs

```diff
@@ -3,11 +3,19 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use std::fmt::Debug;
+use std::ops::Range;
 
 use packed_struct::prelude::*;
 use packed_struct::PackingError;
 use serde::{Deserialize, Serialize};
 
+// Fig 17, section 7.3.4
+pub const HEADER_IMAGE_LENGTH: Range<usize> = 0x20..0x24;
+pub const HEADER_IMAGE_TYPE: Range<usize> = 0x24..0x28;
+pub const HEADER_OFFSET: Range<usize> = 0x28..0x2c;
+pub const HEADER_LOAD_ADDR: Range<usize> = 0x34..0x38;
+pub const HEADER_SIGNATURE: [u8; 4] = *b"cert";
+
```

Comment:
This implementation is not out of line but we have multiple implementations of LPC55 image handling. In addition to [code in this crate](https://github.com/oxidecomputer/lpc55_support/blob/master/lpc55_sign/src/signed_image.rs), the  [lpc55 update-server](https://github.com/oxidecomputer/hubris/blob/master/drv/lpc55-update-server/src/images.rs) and [start up code](https://github.com/oxidecomputer/hubris/blob/master/lib/lpc55-rot-startup/src/images.rs) implement related logic.
Can the low-level image parsing move into `signed_image.rs` or consolidated into a common `image.rs` in this crate?

I would love to consolidate all of the LPC55 image handling into a crate that can also be used in an embedded context. You don't need to go that far here.

Reply:
I agree that it would be nice to consolidate even more code into this, but I am not currently set up to do Hubris development or testing. If you'd like to open issues or PRs for those Hubris bits, that would be great!

> Can the low-level image parsing move into `signed_image.rs` or consolidated into a common `image.rs` in this crate?

In this PR, the low-level signed-image parsing routine is `image_certs_and_sig`, which is already in `signed_image.rs`. That doesn't seem like the right location for these definitions, though, because (most of them) are also used in other contexts (like CRC images, which you could argue should be dropped, but I think that's a separate issue).